### PR TITLE
Vanilla WebComponents implementation

### PR DESCRIPTION
**Pull request by dsolimando**
Hello !

This is my little contribution to this great project.

It's a webcomponents based implementation without any framework or build tools (just polyfills for cross browser support).

### CODE REVIEW

**Comment by cristianbote** [comment]
Hey @dsolimando!

Your submission caught my eye, because it has the `vanilla` in title 😄 . Looking over the code and implementation, I think there are a couple of things, that I had to change as well for my version just so across the implementation there's a similarity in experience and data aggregations.

So, I think, would be best if:

1. You would add at least a`prev 1 next` pagination, and not the lazy loading implementation. Also, the url should reflect the current page. Keep in mind, the requirements states that


2. Make it usable for desktop as well. Restrain the full width design, something like these implementation have:
https://hn.kristoferbaxter.com/
https://vue-hn.now.sh/
https://hnpwa-vanilla.firebaseapp.com/ (this is my version)

3. I would challenge the fact, that this version is not SSR. By design, I think it is. The initial data, is dumped in the server side rendered html. https://github.com/dsolimando/hnpwa-mobileelements/blob/master/shows/__index.html#L76

4. Another issue I noticed is for `Jobs` there's a `null` showing instead of points and user.

Cheers!

**Comment by dsolimando** [comment]
Hello @cristianbote !

Thanks a lot for your suggestions.

I don't quite understand the use of the old pagination pattern in other implementations. If PWA's want to compete with native apps, should they not follow native pattern ?  I've never seen any native app implementing pagination that way.

**Comment by anubhav7495** [comment]
Hi @dsolimando, thanks for your app submission. 

Regarding your queries:

1. Pagination is a better UX pattern in our case.
For reference: [pagination vs Infinite scroll](“UX: Infinite Scrolling vs. Pagination” @101babich https://uxplanet.org/ux-infinite-scrolling-vs-pagination-1030d29376f1) 

2. PWA's are not competing with Native Apps, but with basic web apps.

I would request you to read the submission requirements and modify your app for maintaining consistency with other implementations. 

Thanks

**Comment by dsolimando** [comment]
Hi @anubhav7495 

I'm late in the game so I'll not challenge anything but saying PWA's are not competing with Native Apps is not a valid statement.

All PWA's features are basically a mean to fill the gap with native apps:

 * Offline
 * APPlication shell
 * Push notifications

These features allow to implement the vast majority of the apps existing on different stores.

Every article discussing PWA is comparing their use in place of a native app.

[here](https://medium.com/javascript-scene/native-apps-are-doomed-ac397148a2c0)
[here](https://en.wikipedia.org/wiki/Progressive_web_app)

Just considering PWA as a replacement for a website harms the technology.

Where can I find  the submission requirements?

Thanks

**Comment by anubhav7495** [comment]
https://github.com/tastejs/hacker-news-pwas#specification

**Comment by dsolimando** [comment]
Fine!

It seems every condition is filled!

Thank you!

**Comment by anubhav7495** [comment]
@dsolimando, I have highlighted the things below that need to be worked on before this app can be merged.

1. Each view should have a unique URL, otherwise refreshing the app takes you to the `top stories` view.
2. Pagination instead of the infinite scroll with unique URL for each page for same reasons as above.
3. Comments view should be a separate view with unique URL for each story.
4. `serviceWorkerRegistartion` is logged to the console.
5. Please consider having a `header` instead of a `footer`.
6. Please make the app width responsive on larger screens as it gets difficult to scroll all the way to click on links.
7. `Jobs` view shows `null points by null`, please fix that.
8. Lighthouse report link is missing in `webcomponents-hn.txt`.
9. The App has a lighthouse score of less than 90/100. [WPT LightHouse Report](https://www.webpagetest.org/lighthouse.php?test=170811_31_dfa6a23d4d724aa1839099d9a136e87a&run=1)
10. Please run the WPT tests with Lighthouse report enabled and use those links in the PR
11. Please use the Interactive scores from the lighthouse reports for both `Faster-3G` and `3G-EM`.

You can take a look at below apps for some pointers:
[react-hn](https://react-hn.kristoferbaxter.com/)
[Vanilla-hn](https://hnpwa-vanilla.firebaseapp.com/?page=1)

Thanks.

**Comment by addyosmani** [comment]
Thanks for the thorough review, @anubhav7495. We appreciate your time and patience with addressing the feedback from reviewers here, @dsolimando. Thanks for the submission!

**Comment by dsolimando** [comment]
Hello,

I made the needed corrections.

Thanks for reviewing again 

Damien

**Comment by cristianbote** [comment]
Hey @dsolimando, thanks for addressing the issues. I was looking over your app on my phone, and I noticed I basically can't read the last 2 stories on each page. Check out the screenshot:
image

Also, I would update the SSR to yes, since you are dumping the initial json payload, inside the body. Here https://github.com/dsolimando/hnpwa-mobileelements/blob/master/shows/__index.html#L76

**Comment by dsolimando** [comment]
Hello @cristianbote !

Thanx for your feedbacks.

It's corrected now!

Cheers

Damien

**Comment by addyosmani** [comment]
cc @robdodson as an fyi

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,22 @@
+0 info it worked if it ends with ok
```
Comment (addyosmani) [inline npm-debug.log]: npm-debug.log should not be committed here.

File: site/Gemfile.lock [hunk 2]
```diff
@@ -50,7 +50,7 @@ DEPENDENCIES
   minima (~> 2.0)
 
 RUBY VERSION
-   ruby 2.4.1p111
+   ruby 2.0.0p648
 
 BUNDLED WITH
-   1.15.1
+   1.15.3
```
Comment (addyosmani) [inline site/Gemfile.lock]: Please revert changes to the Gemfile.log. Only changes needed for the implementation you are adding should be in this PR.

File: site/npm-debug.log [hunk 3]
```diff
@@ -0,0 +1,49 @@
+0 info it worked if it ends with ok
```
Comment (addyosmani) [inline site/npm-debug.log]: Similarly, please exclude this npm-debug.log from your PR.

**Comment by dsolimando** [comment]
@addyosmani Thanks for the review!

I did commit this inadvertently in my first PR but did correct after.

It seems good now [here](https://github.com/tastejs/hacker-news-pwas/pull/106/files)

Is the PR correct ?

**Comment by addyosmani** [comment]
@dsolimando Those aren't in the latest commits so we're good on that front. Cheers.

It would probably be best to check if @anubhav7495 has more feedback on the items he raised as we'd want his approval before landing this too.

**Comment by dsolimando** [comment]
@addyosani thanks for the clarification!

@anubhav7495 are you ok with the last version of the app ?

Thank you

**Comment by dsolimando** [comment]
@addyosmany @anubhav7495 a month has elapsed since my PR, do I have a chance to see it reevaluated? 

Thank you

**Comment by anubhav7495** [comment]
@dsolimando Extremely sorry for the delay, got busy with some things. I would take a look by tomorrow and update the status.

Once again, thanks for your contribution.

**Comment by anubhav7495** [comment]
@dsolimando Everything looks good now. Just some minor things and we can land this app.

1. Clicking on any list item opens up the comments view for them, ideally it should open the link associated with it.
2. On scrolling completely down, the pagination button changes it's state to that of a disabled button or link but is still clickable, is that by mistake or design?

Thanks

**Comment by dsolimando** [comment]
@anubhav7495 thanks for the review!

1) Yeah it's not ideal. Actually clicking on the title in the comments view open the HN article. I'l will fix that asap

2) It's by design. The buttons are not disabled, just translucent. This allow the reading the last article of the page.

**Comment by dsolimando** [comment]
Hello @anubhav7495 !

I fixed the 2 points you mentioned in your last review.

I hope all is fine now

Cheers

Damien

**Comment by anubhav7495** [comment]
Hi @dsolimando 

`Ask HN` links are not working. This is because they should always open the comments view which is not the case now. Everything else looks good.

Thanks.

**Comment by dsolimando** [comment]
Hello @anubhav7495 ,

I fixed your last point.

Cheers

**Comment by dsolimando** [comment]
Hello @anubhav7495,

Is it ok for you ?

Thanks a lot

**Comment by anubhav7495** [comment]
@dsolimando Clicking on `Ask HN` links should open the comments view when clicked. It gives a 404 error page right now.

**Comment by addyosmani** [comment]
@dsolimando Verifying that the current behavior around the `Ask HN` still needs some work. Do you think you'll have a chance to update? We're pretty close to this one being done.

**Comment by dsolimando** [comment]
@addyosmani @anubhav7495 Updated!

**Comment by anubhav7495** [comment]
`Ask HN` links in the `Top` or `New` list are still broken. They work fine in the `Ask` list.

**Comment by dsolimando** [comment]
@anubhav7495 sure you have the last version ? Tested in safari ( no SW support) and it works just fine

**Comment by anubhav7495** [comment]
@dsolimando yeah. I checked in incognito.

**Comment by dsolimando** [comment]
Ok sorry It's another bug. I will fix

Thank you

**Comment by dsolimando** [comment]
@anubhav7495 Fixed! Hope this one is the one ;)

**Comment by dsolimando** [comment]
@anubhav7495 @addyosmani ok for you guys?

**Comment by anubhav7495** [comment]
Perfect! Thanks @dsolimando.

**Comment by dsolimando** [comment]
@anubhav7495 @addyosmani Hello guys, Is a merge planned any time soon ?

Cheers

**Comment by dsolimando** [comment]
Maybe for Christmas as a Santa gift ;)

**Comment by dsolimando** [comment]
Thank you very much @addyosmani and @anubhav7495 for your feedbacks. 

Happy new year!

Ho I noticed my scores are missing on hnpwa.com.

**Comment by anubhav7495** [comment]
@dsolimando Made a PR to fix that #135 
@addyosmani Please Merge

**Pull request merged**